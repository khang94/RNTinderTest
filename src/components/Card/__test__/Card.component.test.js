import React from 'react';

import CardComponent from '../Card.component';
import {shallow} from 'enzyme';

jest.mock('../Card.handlers', () => ({
  getUserInformation: jest.fn(),
}));

describe('CardComponent snapshot', () => {
  const props = {
    avatar: 'url/123',
    user: {
      name: {
        first: 'Steve',
        last: 'Job',
      },
      location: {
        street: '',
      },
      phone: '',
      email: '',
      password: '',
    },
  };

  it('Should render snapshot for card view', () => {
    const wrapper = shallow(<CardComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should check initial state', () => {
    const wrapper = shallow(<CardComponent {...props} />);
    const componentState = wrapper.state();
    expect(componentState.activeTab).toEqual('location');
  });

  it('Should active phone tab', () => {
    const wrapper = shallow(<CardComponent {...props} />);
    const instance = wrapper.instance();
    instance.onSelectItem('phone');
    expect(wrapper.state().activeTab).toEqual('phone');
  });
});
