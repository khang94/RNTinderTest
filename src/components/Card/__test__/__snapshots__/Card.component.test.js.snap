// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent snapshot Should render snapshot for card view 1`] = `ShallowWrapper {}`;
